C error when on cast of struct with time.StopWatch to interface.

### Describe the bug

A C error is generated when you attempt to cast a struct that has a `time.StopWatch` field to an interface.

### Reproduction Steps

```v
import time

interface Cursor {
mut:
	blink time.StopWatch
}

struct LineCursor {
__global:
	blink time.StopWatch = time.new_stopwatch(auto_start: true)
}

fn main() {
	_ := Cursor(LineCursor{})
}
```

### Expected Behavior

Expected code to compile and run with no output.

### Current Behavior

```
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:512: warning: WINVER redefined
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:7188: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14929: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14935: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14941: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14949: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14975: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14975: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14995: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14997: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:14997: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:15017: warning: assignment from incompatible pointer type
C:/Users/imado/AppData/Local/Temp/v_0/test.13346500263690300193.tmp.c:15188: error: cannot convert 'struct time__StopWatchOptions *' to 'struct time__StopWatchOptions'
builder error: 
==================
C error. This should never happen.
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.2 ae5b4bb.6b79b2a

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.2 ae5b4bb.6b79b2a
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian,

getwd: C:\Windows\System32
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-10-03 14:54:04

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: 0.4.2-22-g6b79b2aa
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```
> [!NOTE]
> You can vote for this issue using the 👍 reaction. More votes increase the issue's priority for developers.
>
> Take into account that only the 👍 reaction counts as a vote.
> Only reactions to the issue itself will be counted as votes, not comments.

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -323,6 +323,11 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 }
 
 fn (mut g Gen) zero_struct_field(field ast.StructField) bool {
+	old_inside_cast_in_heap := g.inside_cast_in_heap
+	g.inside_cast_in_heap = 0
+	defer {
+		g.inside_cast_in_heap = old_inside_cast_in_heap
+	}
 	sym := g.table.sym(field.typ)
 	field_name := if sym.language == .v { c_name(field.name) } else { field.name }
 	if sym.info is ast.Struct {

--- a/vlib/v/tests/cast_to_interface_with_struct_field_default_test.v
+++ b/vlib/v/tests/cast_to_interface_with_struct_field_default_test.v
@@ -1,0 +1,19 @@
+import time
+
+interface Cursor {
+mut:
+	blink time.StopWatch
+}
+
+struct LineCursor {
+__global:
+	blink time.StopWatch = time.new_stopwatch(auto_start: true)
+}
+
+fn test_cast_to_interface_with_struct_field_default() {
+	c := Cursor(LineCursor{})
+	time.sleep(2 * time.millisecond)
+	duration := c.blink.elapsed().milliseconds()
+	println(duration)
+	assert duration > 0
+}

